src/Containers/Microsoft.NET.Build.Containers/ImageConfig.cs

```diff
@@ -53,21 +61,30 @@ internal ImageConfig(JsonNode config)
         _exposedPorts = GetExposedPorts();
         _environmentVariables = GetEnvironmentVariables();
         _rootFsLayers = GetRootFileSystemLayers();
-        _architecture = GetArchitecture();
-        _os = GetOs();
         _history = GetHistory();
         _user = GetUser();
         _newEntrypoint = GetEntrypoint();
         _newCmd = GetCmd();
+
+        // we don't currently allow setting these properties - they are tied to the base image
+        // and represent the platform that the base image (and this anything that derives from this image) was built for
+        _architecture = GetArchitecture();
+        _os = GetOs();
+        _variant = GetVariant();
+        _osVersion = GetOsVersion();
+        _osFeatures = GetOsFeatures();
+
     }
 
-    // Return values from the base image config.
     internal string? GetUser() => _config["config"]?["User"]?.ToString();
     internal string[]? GetEntrypoint() => _config["config"]?["Entrypoint"]?.AsArray()?.Select(node => node!.GetValue<string>())?.ToArray();
     private string[]? GetCmd() => _config["config"]?["Entrypoint"]?.AsArray()?.Select(node => node!.GetValue<string>())?.ToArray();
     private List<HistoryEntry> GetHistory() => _config["history"]?.AsArray().Select(node => node.Deserialize<HistoryEntry>()!).ToList() ?? new List<HistoryEntry>();
     private string GetOs() => _config["os"]?.ToString() ?? throw new ArgumentException("Base image configuration should contain an 'os' property.");
     private string GetArchitecture() => _config["architecture"]?.ToString() ?? throw new ArgumentException("Base image configuration should contain an 'architecture' property.");
+    private string? GetVariant() => _config["variant"]?.ToString();
+    private string? GetOsVersion() => _config["os.version"]?.ToString();
+    private string[]? GetOsFeatures() => _config["os.features"]?.AsArray()?.Select(node => node!.GetValue<string>())?.ToArray();
```

Comment:
In order to check platform compatibility, we needed to read the various fields that are part of the 'platform information' from the base image config. We had some of them before, but we needed these to complete the picture.